### PR TITLE
[Cookbook] Add Electron + Vue cookbook

### LINKED ISSUE #1573: [Cookbook Idea] Building a desktop application with Electron and Vue
# Vue + Electron Cookbook Recipe

Is there interest for a recipe on building desktop apps with Electron and Vue? Here's a brief overview of how the recipe would follow the template:

**Solve a specific, common problem**: Build a desktop application's UI with Vue

**Start with the simplest possible example**: I'm thinking of doing a markdown editor or tic tac toe but I'm open to suggestions

**Introduce complexities one at a time**: Go over setting up an Electron app ready for Vue with electron-vue, the file structure of the project, then write tic tac toe or a markdown editor in Vue

**Link to other docs, rather than re-explaining concepts**: Link to Electron docs and [electron-vue](https://simulatedgreg.gitbooks.io/electron-vue/content/en/) docs. 

**Explain the process, rather than just the end result**: I plan on briefly going over Electron (just enough to go through the tutorial and link to docs for the more info), explaining how Vue fits in, and how we can write Vue for Electron apps.

**Explain the pros and cons of your strategy, including when it is and isn’t appropriate**: Electron and Vue are convenient and great for quickly prototyping cross-platform desktop apps. However, Electron lacks in performance compared to native desktop apps. Electron is able render Vue components because Electron uses Chromium (a web browser) on its front end.

### PR DESCRIPTION
**Pull request by konaraddi**
Closes #1573 and improves upon #1611 

For the Base Example section, I had to deviate from the guidelines a bit to provide enough steps for the reader to follow. Please let me know if and what changes need to be made. Thanks!

### CODE REVIEW

**Comment by sdras** [comment]
Thanks! I'll take a look soon

**Comment by konaraddi** [comment]
Hi @sdras , would you be able to review the cookbook? If not, no worries (I totally understand if you're time is better spent elsewhere!)

**Comment by sdras** [comment]
I know you've waited a bit and apologize on the delay! but I'm still on honeymoon until later this week. I will get back to you next week at the latest. Thanks for the ping.

**Comment by konaraddi** [comment]
Thanks! Enjoy your honeymoon!

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "vuejs.org",
   "private": true,
   "hexo": {
-    "version": "3.7.0"
+    "version": "3.7.1"
```
Comment (phanan) [inline package.json]: Is this change necessary for the PR?

File: src/v2/cookbook/electron-vue-cookbook.md [hunk 2]
```diff
@@ -0,0 +1,704 @@
...
+
+  <!--Use the running variable to disable buttons-->
+  <button @click="startTimer" :disabled="running">Start</button>
+  <button @click="resetTimer" :disabled="!running">Reset</button>
+
+</div>    
+</template>
+
+<script>
+export default {
```
Comment (phanan) [inline src/v2/cookbook/electron-vue-cookbook.md]: IMHO this code example is a bit too long, which makes it harder for 1) readers to follow and 2) maintainers to sync between the markdown and the raw scripts. Can we use something shorter e.g. (OTOMH) the todo app?

File: package.json (hunk 1 above)
Comment (konaraddi) [reply package.json]: Oops, I bumped it back down in 28d53af. I don't remember specifically wanting to bump up the version of hexo (I haven't use hexo elsewhere); might be from running `npm install`. Anyways, I'll be sure to commit only the necessary files next time. Thanks!

File: src/v2/cookbook/electron-vue-cookbook.md (hunk 2 above)
Comment (konaraddi) [reply src/v2/cookbook/electron-vue-cookbook.md]: I considered the todo app but went with the timer because building a timer 
* is quick to build and easy to understand what the code is doing 
* is a small amount of code (and less code than results for todo lists built with Vue on GitHub)

A timer lets us focus on the Electron part of the cookbook. I think with a todo list I'd be spending most of the tutorial on how to build a todo list in Vue, and many existing tutorials for Vue involve building todo lists.

**Comment by konaraddi** [comment]
Should I copy and paste the `package-lock.json` from `vuejs/vuejs.org` to `konaraddi/vuejs.org` to fix the conflict?

**Comment by konaraddi** [comment]
@sdras @phanan I updated the cookbook for Vue CLI 3, can you have a look whenever you have time?

File: src/v2/cookbook/electron-vue-cookbook.md (hunk 2 above)
Comment (phanan) [reply src/v2/cookbook/electron-vue-cookbook.md]: I still maintain that the current code is too long for a cookbook. I personally find it hard to follow, and would like to think that defeats the purpose. 
Btw (and I missed this before), it would be far better IMHO if we don't need to rely on a third party boilerplate (simulatedgreg/electron-vue) for this.

File: src/v2/cookbook/electron-vue-cookbook.md [hunk 3]
```diff
@@ -0,0 +1,705 @@
...
+};
+</script>
+
+<style>
+</style>
+```
+
+Next, let's add buttons for setting the timer's default value. We'll need to add variables for the default minutes and seconds too. When we click the Reset button, our timer will set itself to the default minutes and seconds variables. Make the changes indicated in the comments below.
+
+```vue<template>
```
Comment (sdras) [inline src/v2/cookbook/electron-vue-cookbook.md]: This needs to be on a separate line, like:
```
... vue
<template>
```
(can't put backticks in because it would break, but hopefully you get the drift)

File: src/v2/cookbook/electron-vue-cookbook.md (hunk 2 above)
Comment (sdras) [reply src/v2/cookbook/electron-vue-cookbook.md]: I agree with @phanan, I think this cookbook entry is valuable but it strays from the point sometimes. In this cookbook you probably shouldn't be explaining computed properties and refactoring. It would be better to show  a bit of the code, and have the full thing hosted elsewhere, and then build on that by only what's changed with one line for context and an ellipsis [...] to convey there's more

Otherwise, you would have to go the other route with a much smaller example. You can make a really small [Vue to do](https://codepen.io/yuugou727/pen/NpQpor) that's more immediately legible than the timer.

File: src/v2/cookbook/electron-vue-cookbook.md [hunk 4]
```diff
@@ -0,0 +1,705 @@
+---
+title: Building Desktop Apps with Electron and Vue
+type: cookbook
+order: 13
```
Comment (sdras) [inline src/v2/cookbook/electron-vue-cookbook.md]: I'm pretty sure we're past 13 now, this should be updated

**Comment by sdras** [comment]
Any news on this one? Would you like some help getting it done?

In the future, we'll be using Google Docs for all these changes so it can go a little faster.

**Comment by konaraddi** [comment]
Hi Sarah, thanks for following up! Currently busy with school work and other commitments but I can work on a rewrite in June. Does that work?

**Comment by sdras** [comment]
Yep, that works! Thank you for the update @konaraddi :)
